lib/rspec-puppet/support.rb

```diff
@@ -13,7 +13,9 @@ def load_catalogue(type)
       code = [import_str, pre_cond, test_manifest(type)].join("\n")
       node_name = nodename(type)
 
-      catalogue = build_catalog(node_name, facts_hash(node_name), code)
+      hiera_config_value = self.respond_to?(:hiera_config) ? hiera_config : nil
+
+      catalogue = build_catalog(node_name, facts_hash(node_name), hiera_config_value, code)
```

Comment:
The trick seems to be that adding the configuration as additional argument will change the cache key used in build_catalog(). Without this, the cached catalog is returned unchanged.